### PR TITLE
[TW-1276] Spectral parameter and property descriptions

### PR DESCRIPTION
**Pull request by MatthewPinsker-postman**
* Changed `input` parameter name to `targetVal` to match the boilerplate function in the UI. Updated all occurrences.
* Clarified how to use the custom function parameters.
* Improved descriptions of the `targetVal`, `options`, and `context` parameters. Also improved the description of the `path` property.
* Changed the "Spectral function object properties" header to "Exporting your custom function". We need to make this section generic because ES6 (coming soon) uses different syntax.
* Added links to relevant sections for more information about Spectral properties.
* Some wording and style changes.

### CODE REVIEW

**Review by mark-anthony-postman — APPROVED** [review]
LGTM! 🌟

File: src/pages/docs/api-governance/configurable-rules/spectral.md [hunk 1]
```diff
@@ -202,49 +206,45 @@ Use the following properties to write the return statement in your custom functi
...
+`path` | <p>An optional path to an element in the document that triggers the rule violation. You can add a path to the `path` property to investigate sub-elements of the `targetVal` value or other locations in the document. The path must be an array of strings, such as `["paths", "/resources", "get", "responses", "306"]`.</p> <p>If you use the `path` property, you must also add the [`context` parameter](#spectral-function-parameters) to your function. This enables you to access the `targetVal` value or other locations in the document.</p> <p>If you don't add the `path` property, the default path is the `targetVal` value.</p> <p>When targeting sub-elements of the `targetVal` value, add `...context.path` to the beginning of the path. This enables you to append a path to `targetVal`. For example, `path: [...context.path, "a", "custom", "path"]`.</p>
```
Comment (arno-di-loreto) [inline src/pages/docs/api-governance/configurable-rules/spectral.md:209]: It's the `targetVal` path you can access, not its value:
 
 This enables you to access the `targetVal` ~~value~~ path or other locations in the document.
 
 With more context, that could also give something like this:

This enables you to other locations in the document or access the `targetVal` path, which is necessary when investigating sub-elements of `targetVal`.

File: src/pages/docs/api-governance/configurable-rules/spectral.md [hunk 2]
```diff
@@ -179,20 +179,24 @@ rules:
...
+`options` | <p>The second parameter you can add to your function. This is the optional value of the [`then.functionOptions` property](#spectral-rule-properties). Add this parameter to your function if your function expects options.</p>
+`context` | <p>The third parameter you can add to your function. You can use this optional parameter to access properties about the context in which the custom function is called. For example, you can access sub-elements of `targetVal` or other locations in the document. These properties are as follows: </p><ul><li>`path` - The path to `targetVal` in the form of an array of strings. For example, `["paths", "/resources", "get", "responses", "306"]`. To learn how to use it, see [Spectral function return statement properties](#spectral-function-return-statement-properties). </li><li>`document` - The document you're attempting to lint.</li><li>`rule` - The rule that's using the function.</li><li>`documentInventory` - Provides access to resolved and unresolved documents, the `$ref` resolution graph, and other advanced properties.</li></ul>
```
Comment (arno-di-loreto) [inline src/pages/docs/api-governance/configurable-rules/spectral.md:196]: Users don't need `context` to access sub-elements of `targetVal`; they need it to access its path to build the paths to the faulty sub-elements, as explained in the description of the `path` field in _Spectral function return statement properties_ section.

For example, you can access ~~sub-elements of~~ `targetVal` **path** or other locations in the document.

If a longer explanation is needed, we can copy/paste what is said in the description of the `path` field in _Spectral function return statement properties_ section.

For example, you can access other locations in the document or get the `targetVal` path, which is necessary when investigating sub-elements of `targetVal`.

File: src/pages/docs/api-governance/configurable-rules/spectral.md (hunk 1 above)
Comment (arno-di-loreto) [reply src/pages/docs/api-governance/configurable-rules/spectral.md:209]:
```suggestion
`path` | <p>An optional path to an element in the document that triggers the rule violation. You can add a path to the `path` property to investigate sub-elements of the `targetVal` value or other locations in the document. The path must be an array of strings, such as `["paths", "/resources", "get", "responses", "306"]`.</p> <p>If you use the `path` property, you must also add the [`context` parameter](#spectral-function-parameters) to your function. This enables you to access other locations in the document or access the `targetVal` path, which is necessary when investigating sub-elements of targetVal.</p> <p>When targeting sub-elements of the `targetVal` value, add `...context.path` to the beginning of the path. This enables you to append a path to `targetVal`. For example, `path: [...context.path, "a", "custom", "path"]`.</p><p>If you don't add the `path` property, the default path is the `targetVal` value.</p> 
```

File: src/pages/docs/api-governance/configurable-rules/spectral.md [hunk 3]
```diff
@@ -258,11 +258,11 @@ module.exports = notInEnumeration;
 
 ### Example: Rule that uses a custom function
```
Comment (arno-di-loreto) [inline src/pages/docs/api-governance/configurable-rules/spectral.md:259]: In this section the function name to use is `not_in_enumeration` (hence the function filename) and not `notInEnumeration` (the name of the exported function in the function file).

File: src/pages/docs/api-governance/configurable-rules/spectral.md [hunk 4]
```diff
@@ -258,11 +258,11 @@ module.exports = notInEnumeration;
...
 
-The following Spectral document has a rule named `http-status-obsolete` that uses a function named `notInEnumeration`. The function is in a file named `not_in_enumeration`, which is defined using the **Name** field when you [create a custom function](/docs/api-governance/configurable-rules/configuring-custom-governance-functions/#adding-a-custom-function).
+The following Spectral document has a rule named `http-status-obsolete` that uses a custom function named `notInEnumeration`. The custom function is in a file named `not_in_enumeration`, which is defined using the **Name** field when you [create a custom function](/docs/api-governance/configurable-rules/configuring-custom-governance-functions/#adding-a-custom-function).
```
Comment (arno-di-loreto) [inline src/pages/docs/api-governance/configurable-rules/spectral.md:261]: Trying to rephrase to clarify `not_in_enumeration` vs `notInEnumeration`.

The following Spectral document has a rule named `http-status-obsolete` that uses the `not_in_enumeration` custom function. The custom function name is the one set in the **Name** field when you [create a custom function](/docs/api-governance/configurable-rules/configuring-custom-governance-functions/#adding-a-custom-function) and not the name of the exported function (`notInEnumeration` in that case).

File: src/pages/docs/api-governance/configurable-rules/spectral.md [hunk 5]
```diff
@@ -179,20 +179,24 @@ rules:
...
-`context` | This optional parameter is used in advanced use cases where you need to investigate several elements. You can use the parameter to access properties about the function. These properties are as follows: <br><ul><li>`path` - The `given` [JSON Path Plus expression](#json-path-and-json-path-plus) pointing to `input`. </li><li>`document` - The document you're attempting to lint.</li><li>`rule` - The rule that's using the function.</li><li>`documentInventory` - Provides access to resolved and unresolved documents, the `$ref` resolution graph, and other advanced properties.</li></ul>
+`targetVal` | <p>**Required**. The first parameter you must add to your function. This can be any data type, such as a string or array. This is the value that the [`given` property](#spectral-rule-properties) returns. The rule tests the value of `targetVal` using your custom function.</p> <p>If you also define a value for the [`then.field` property](#spectral-rule-properties) in your rule, `targetVal` is the value returned by the `given` path appended with `then.field`.</p>
```
Comment (arno-di-loreto) [inline src/pages/docs/api-governance/configurable-rules/spectral.md:194]: Maybe worth mentioning/clarifying a common false assumption about targetVal and how a custom function is executed:  People often think that `targetVal` will be the list of all elements found by the given path(s), which is not the case. If the given path(s) find multiple values, the function will be called one time for each. So `targetVal` will be one of the items found by the given path(s).

For instance, if the given path targets all query parameters and if 10 query parameters are defined in the document, the custom function will be called 10 times. On each execution, targetVal will hold one query parameter definition.

File: src/pages/docs/api-governance/configurable-rules/spectral.md (hunk 1 above)
Comment (MatthewPinsker-postman) [reply src/pages/docs/api-governance/configurable-rules/spectral.md]: This is now updated.

I reorganized the content to first introduce the `path` property, explain any prerequisites, and the default value. Then I broke up its usage into two sections: how to investigate other elements, and how to investigate sub-elements of `targetVal`.

File: src/pages/docs/api-governance/configurable-rules/spectral.md (hunk 2 above)
Comment (MatthewPinsker-postman) [reply src/pages/docs/api-governance/configurable-rules/spectral.md]: This is now updated.

File: src/pages/docs/api-governance/configurable-rules/spectral.md (hunk 4 above)
Comment (MatthewPinsker-postman) [reply src/pages/docs/api-governance/configurable-rules/spectral.md]: This is now updated.

I updated the section to clarify that the custom function filename is added to the custom rule, and the file contains the custom function.

To borrow similar terms from the [Spectral docs](https://github.com/stoplightio/spectral/blob/develop/docs/guides/5-custom-functions.md), the LC docs refer to the exported function as the "custom function", and the container you add to `then.property` in the custom rule as the "custom function file". Maybe we should be using different terms in the LC docs. Please let me know if we need to adjust.

File: src/pages/docs/api-governance/configurable-rules/spectral.md (hunk 5 above)
Comment (MatthewPinsker-postman) [reply src/pages/docs/api-governance/configurable-rules/spectral.md:194]: Let's address this in a separate ticket/PR. I'll create at ticket for tracking.

File: src/pages/docs/api-governance/configurable-rules/spectral.md (hunk 1 above)
Comment (MatthewPinsker-postman) [reply src/pages/docs/api-governance/configurable-rules/spectral.md]: This is now fixed. Thanks!

**Review by arno-di-loreto — APPROVED** [review]

**Pull request merged**